chroma-agent/chroma_agent/action_plugins/device_plugin.py

```diff
@@ -47,5 +54,37 @@ def trigger_plugin_update(agent_daemon_context, plugin_names):
         agent_daemon_context.plugin_sessions[plugin_name]._plugin.trigger_plugin_update = True
 
 
-ACTIONS = [device_plugin, trigger_plugin_update]
+@agent_daemon_startup_function()
+def initialise_block_device_drivers():
+    """
+    When the agent is run we want to allow block devices to do any initialization that they might need, this function
+    may also be called by the manager.
+    """
+    console_log.info("Initialising drivers for block device types")
+    for cls in util.all_subclasses(BlockDevice):
+        error = cls.initialise_driver(config.profile_managed)
+
+        if error:
+            return agent_error(error)
+
+    return agent_result_ok
+
+
+@agent_daemon_teardown_function()
+def terminate_block_device_drivers():
```

Comment:
Both initialize_block_device_drivers and terminate_block_device_drivers have the same structure. Do you think it's worth making a common function between the two and then specifying which operation to take?